docs/source/nodes_and_pipelines/pipeline_introduction.md

```diff
@@ -1,26 +1,30 @@
 # Pipelines
 
-We previously introduced [Nodes](./nodes.md) as building blocks that represent tasks, and can be combined in a pipeline to build your workflow. A pipeline organises the dependencies and execution order of your collection of nodes, and connects inputs and outputs while keeping your code modular. The pipeline resolves dependencies to determine the node execution order, and does *not* necessarily run the nodes in the order in which they are passed in.
+A pipeline is a set of [Nodes](./nodes.md) that will be automatically resolved after pipeline creation to determine the node execution order based on outputs and inputs of each node.
```

Comment:
Not sure if this sentence reads very clear to me. I don't really mind the introduction from before but simplified just a little bit?